### PR TITLE
update naming conventions

### PR DESCRIPTION
**Pull request by jonathantneal**
In an effort to improve our RFC process, this PR simplifies the current file structure.

1. Automatically adds the `0000` prefix before proposals after they are merged.
2. Renames the `active-rfc` directory to `proposals`, and the `000-template.md` file to `template.md`
3. Improves the `README.md` and `template.md` files to match these conventions.

- This also formats `README.md` to 80-characters per line, which was present in three-quarters of the file.

This PR supersedes https://github.com/withastro/rfcs/pull/72

### CODE REVIEW

File: .github/format-rfcs.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import * as fs from 'node:fs/promises'
+
+const dir = new URL('../proposals/', import.meta.url)
```
Comment (boehs) [inline .github/format-rfcs.mjs:3]: I think `active` was better. If it is in that directory, it is no longer proposed and actually to be implemented.

File: README.md [hunk 2]
```diff
@@ -2,143 +2,171 @@
...
 
-## The RFC life-cycle
+## The RFC Life-Cycle
 
 An RFC goes through the following stages:
 
 - **Pending:** when the RFC is submitted as a PR.
 - **Active:** when an RFC PR is merged and undergoing implementation.
-- **Landed:** when an RFC's proposed changes are shipped in an actual release.
+- **Landed:** when an RFC’s proposed changes are shipped in an actual release.
```
Comment (boehs) [inline README.md:21]: Was this intentional?

File: README.md [hunk 3]
```diff
@@ -2,143 +2,171 @@
...
+not these changes should be made.
 
 ## Gathering feedback before submitting
 
-It's often helpful to get feedback on your concept before diving into the
-level of API design detail required for an RFC. **You may open an
-issue on this repo to start a high-level discussion**, with the goal of
-eventually formulating an RFC pull request with the specific implementation
-design.
+It’s often helpful to get feedback on your concept before diving into the level
```
Comment (boehs) [inline README.md:78]: Same here

File: README.md [hunk 4]
```diff
@@ -2,143 +2,171 @@
...
+anybody is currently working on it.
 
-Modifications to active RFC's can be done in followup PR's. We strive
-to write each RFC in a manner that it will reflect the final design of
-the feature; but the nature of the process means that we cannot expect
-every merged RFC to actually reflect what the end result will be at
-the time of the next major release; therefore we try to keep each RFC
-document somewhat in sync with the language feature as planned,
-tracking such changes via followup pull requests to the document.
+Modifications to active RFC’s can be done in followup PR’s. We strive to write
```
Comment (boehs) [inline README.md:135]: This quotation mark is very clearly driving me insane!!!

**Review by boehs — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (jonathantneal) [reply README.md]: That’s an apostrophe. Many computers make this mark automatically. Most browsers search for all of them when you search for any of them.

File: README.md (hunk 2 above)
Comment (jonathantneal) [reply README.md:21]: Yes. That’s an apostrophe.

File: README.md (hunk 3 above)
Comment (jonathantneal) [reply README.md:78]: https://www.fileformat.info/info/unicode/char/2019/index.htm
https://dictionary.cambridge.org/us/grammar/british-grammar/possession-john-s-car-a-friend-of-mine

**Comment by boehs** [comment]
> That’s an apostrophe. Many computers make this mark automatically. Most 
> browsers search for all of them when you search for any of them.
Incredible, the things you learn

File: README.md (hunk 4 above)
Comment (boehs) [reply README.md]: Incredible!

File: .github/format-rfcs.mjs (hunk 1 above)
Comment (jonathantneal) [reply .github/format-rfcs.mjs:3]: The reason this uses `proposals` instead of `active` is so that a `landed` proposal does not change URLs.

**Pull request merged**